scripts/templates/create-component/EmptyState.mustache

```diff
@@ -1,20 +1,20 @@
-import { I{{componentName}}Props, I{{componentName}}ViewProps } from './{{componentName}}.types';
-import { BaseState } from '../../utilities/BaseState';
+import { useState } from 'react';
+import { I{{componentName}}Component, I{{componentName}}Props, I{{componentName}}ViewProps } from './{{componentName}}.types';
+import { useControlledState } from '../../Foundation';
 
-// Internal state will most likely include a subset of your ViewProps. This template just equates them to start with.
-export type I{{componentName}}State = I{{componentName}}ViewProps;
+export const use{{componentName}}State: I{{componentName}}Component['state'] = (
+  props: Readonly<I{{componentName}}Props>
+): I{{componentName}}ViewProps => {
+  const [status] = useState('State Text');
+  const [text] = useControlledState(props, 'text', {
+    defaultPropName: 'defaultText',
+    defaultPropValue: 'Default Text'
+  });
 
-export class {{componentName}}State extends BaseState<I{{componentName}}Props, I{{componentName}}ViewProps, I{{componentName}}State > {
-  constructor(props: {{componentName}}State['props']) {
-    super(props, {
-      // Mark controlledProps to ensure that they get priority when provided as a component prop.
-      // For props not marked controlled, component state will get priority over component props.
-      controlledProps: ['text']
-    });
-
-    this.state = {
-      text: props.defaultText || 'Default Text',
-      status: 'State Text'
-    };
-  }
-}
+  const viewProps: I{{componentName}}ViewProps = {
```

Comment:
Do you want to add a super basic useState here and have a callback passed out? It might be unnecessary boilerplate but it could be useful as an example or jumping off point maybe.

Reply:
There is one on line 8.

Reply:
I'm not sure how the callback would be passed out? I don't think createComponent would be using callbacks...

Reply:
I'm surprised that I missed that! I see you using it now.

The callback is part of viewProps, no? Like an onclick or onChange. That's the only way a state change is going to occur.

Reply:
I'll make a few more changes, add a Button, and create a callback that changes state.